src/pages/LoginPage.js

```diff
@@ -1,32 +1,53 @@
 import React from 'react';
-import {Link} from "react-router-dom";
-
 import NavBar from '../components/Navbar'
+import EventService from "./EventService";
+
+class LoginPage extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {email: '', password: '', isAuthenticated: false};
+    }
+
+    onPasswordChange(e) {
+        this.setState({password: e.target.value});
+    }
+
+    onEmailChange(e) {
+        this.setState({email: e.target.value});
+    }
 
+    handleSubmit(event){
+        event.preventDefault();
+        EventService.auth.userLogin(this.state, (authSuccess) => {
+            this.setState({isAuthenticated: authSuccess});
+            if(authSuccess){
+                this.props.history.push('/admin', this.state);
```

Comment:
Now, that I think of it maybe we do need to do res.cookie on backend as well to persist jwt. 
Or you can use localStorage to persist jwt, which should be enough for the scope of this course.

Reply:
axios is setting the default headers with the token. Wouldn't this sufficiently 'persist' the jwt until the user logs out?

Reply:
setting the default header is the same as storing memory, Once user refresh it's gone I think.

Reply:
I was working on login as well, I wrote some functions that could be useful for this
```javascript
export const login = () => {
    localStorage.setItem(TOKEN_KEY, 'TestLogin');
}

export const logout = () => {
    localStorage.removeItem(TOKEN_KEY);
}

export const isLogin = () => {
    if (localStorage.getItem(TOKEN_KEY)) {
        return true;
    }

    return false;
}
```

Reply:
I see, so the main idea would be to store the jwt in localStorage, then retrieve, and set the auth header at request time

Reply:
The latest commit should solve this issue, take another look @leogoldberg  @JasonXD-CS